src/create_disk.sh

```diff
@@ -408,16 +521,26 @@ s390x)
     # See https://github.com/coreos/ignition-dracut/issues/84
     # A similar hack is present in https://github.com/coreos/coreos-assembler/blob/main/src/gf-platformid#L55
     echo "$(grep options $blsfile | cut -d' ' -f2-) ignition.firstboot" > $tmpfile
-
-    # ideally we want to invoke zipl with bls and zipl.conf but we might need
-    # to chroot to $rootfs/ to do so. We would also do that when FCOS boot on its own.
-    # without chroot we can use --target option in zipl but it requires kernel + initramfs
-    # pair instead
-    zipl --verbose \
-        --target $rootfs/boot \
-        --image $rootfs/boot/"$(grep linux $blsfile | cut -d' ' -f2)" \
-        --ramdisk $rootfs/boot/"$(grep initrd $blsfile | cut -d' ' -f2)" \
-        --parmfile $tmpfile
+    kernel=$rootfs/boot/"$(grep linux $blsfile | cut -d' ' -f2)"
+    initrd=$rootfs/boot/"$(grep initrd $blsfile | cut -d' ' -f2)"
+    if [[ ${secure_execution} -eq 1 ]]; then
+        propogate_luks_config "${secure_execution_bootfs_key}" \
+                              "${secure_execution_bootfs_label}"
+        propogate_luks_config "${secure_execution_rootfs_key}" \
+                              "${secure_execution_rootfs_label}"
+        sdboot=$rootfs/se/sd-boot
+        regenerate_initramfs $initrd $secure_execution_initrd
+        genprotimg -i $kernel -r $secure_execution_initrd -p $tmpfile -k $secure_execution_hostkey --no-verify -o $sdboot
+        rm -f $secure_execution_initrd
+        # zipl SecureExecution image
+        zipl --verbose --target $rootfs/se --image $sdboot
+    else
+        # ideally we want to invoke zipl with bls and zipl.conf but we might need
+        # to chroot to $rootfs/ to do so. We would also do that when FCOS boot on its own.
+        # without chroot we can use --target option in zipl but it requires kernel + initramfs
+        # pair instead
+        zipl --verbose --target $rootfs/boot --image $kernel --ramdisk $initrd --parmfile $tmpfile
+    fi
```

Comment:
Hmm, this is now a third implementation of this logic. WDYT about using `rdcore zipl` here instead? I think it should work if we mount things the right way before calling it.

Reply:
yep, but i guess we first need new `coreos-installer.rpm`